awsapilib/console/console.py

```diff
@@ -464,32 +445,31 @@ def _validate_response(response):
         return success
 
     def _resolve_account_type(self, email):
-        response = self._resolve_account_type_response(email, {'hashArgs': '#a'})
+        response = self._resolve_account_type_response(email)
         success = self._validate_response(response)
         if not success:
             raise UnableToResolveAccount(f'Failed to resolve account type with response: {response.text} '
                                          f'and status code {response.status_code}')
         self.logger.debug(f'Resolved account type successfully with response :{response.text}')
         return success
 
-    def _resolve_account_type_response(self, email, extra_parameters=None, session=None):
+    def _resolve_account_type_response(self, email, session=None):
         session_ = session if session else self.session
+
         parameters = {'action': 'resolveAccountType',
-                      'email': email}
-        if extra_parameters:
-            parameters.update(extra_parameters)
-        _ = session_.get(self._console_home_url, params=parameters)
-        parameters.update({'csrf': session_.cookies.get('aws-signin-csrf', path='/signin')})
+                      'email': email,
+                      'csrf': session_.cookies.get('aws-signin-csrf', path='/signin')}
         response = session_.post(self._signin_url, data=parameters)
         self.logger.debug('Getting the resolve account type captcha.')
         parameters = self._update_parameters_with_captcha(parameters, response)
         return session_.post(self._signin_url, data=parameters)
 
-    def get_mfa_type(self, email):
+    def get_mfa_type(self, email, redirect_url_for_mfa):
```

Comment:
This poses a problem since `get_mfa_type` is a public method. A user trying to get the mfa type would not have a redirect url for mfa. We need to think about this a bit on how the interface can be presented. I am guessing that since you implemented it like this then the redirect url is no longer computable and it is provided by the OIDC object in calls before. If that is indeed the case then this method cannot be public any more since it cannot possbily provide value to a user.

Reply:
I reintroduced it, since the URL can still be synthesized.